addons/website_sale_coupon/tests/test_apply_pending_coupon.py

```diff
@@ -0,0 +1,91 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import TransactionCase, tagged
+from odoo.addons.website.tools import MockRequest
+from odoo.addons.website_sale_coupon.controllers.main import WebsiteSale
+
+import pprint
+
+pp = pprint.PrettyPrinter(indent=4)
+
+
+@tagged('-at_install', 'post_install')
+class TestSaleCouponApplyPending(TransactionCase):
```

Comment:
inherit `TestSaleCouponProgramNumbers` here and use already existing program and product